src/vivarium/framework/results/manager.py

```diff
@@ -130,6 +159,8 @@ def register_observation(
         )
 
     def _add_resources(self, target: List[str], target_type: str):
+        if not len(target):
+            return  # do nothing on empty lists
```

Comment:
Curious on checks for empty lists/dataframes/etc and good practices for that.

Reply:
Here, it's not strictly necessary, but it's a low-cost check as most `target` lists will be of length 1. Good practice is to consider the tradeoff of the performance hit of the check vs the benefit.